ui/controllers/userController.js

```diff
@@ -0,0 +1,109 @@
+        .withMessage('Passwords don\'t match'),
+    sanitizeBody('username').trim().escape(),
+    sanitizeBody('email').trim().escape(),
+    function(request, response, next) {
+        let errors = validationResult(request)
+        if (!errors.isEmpty()) {
+            response.status(200).render('register', { title: 'Register', user: request.body, errors: errors.array() })
+        } else {
+            const userData = {
+                username: request.body.username,
+                display_name: request.body.username,
+                email: request.body.email,
+                password: request.body.password
+            }
+            User.create(userData, (error, user) => {
+                if (error) {
+                    if (error.code === 11000) {
+                        // error.code indicates a user was found
+                        let error = [{ msg: 'Username or email already exists' }]
+                        response.status(200).render('register', { user: request.body, errors: error })
+                    } else {
+                        logger.log('error', error)
+                        return next(error)
+                    }
+                } else {
+                    // save the session for two weeks if the user checked the 'remember' box
+                    if (request.body.remember) request.session.cookie.maxAge = 14 * 24 * 3600000
+                    request.session.userId = user._id
+                    response.status(200).redirect('/profile')
+                }
+            })
+        }
+    }
+]
+
+exports.post_login = [
+    body('username').trim().isLength({ min: 1 }).withMessage('Username must be specified'),
+    body('password').exists().withMessage('Password is required'),
+    sanitizeBody('email').trim().escape(),
+    function(request, response, next) {
+        let errors = validationResult(request)
+        if (!errors.isEmpty()) {
+            response.status(200).render('login', { title: 'Login', errors: errors.array() })
+        } else {
+            User.authenticate(request.body.username, request.body.password, function(error, user) {
+                if (error) {
+                    if (error.status === 401) {
+                        let error = [{ msg: 'Incorrect email or password' }]
+                        response.status(200).render('login', { title: 'Login', errors: error })
+                    } else {
+                        logger.log('error', error)
+                        return next(error)
+                    }
+                } else {
+                    // save the session for two weeks if the user checked the 'remember' box
+                    if (request.body.remember) request.session.cookie.maxAge = 14 * 24 * 3600000
+                    request.session.userId = user._id
+                    response.redirect('/profile')
+                }
+            })
+        }
+    }
+]
```

Comment:
Same comment about this handler, break out logic into separate controller and include tests